stl/inc/xutility

```diff
@@ -2842,15 +2842,19 @@ namespace ranges {
     public:
         using _Not_quite_object::_Not_quite_object;
 
-        template <input_or_output_iterator _It, sentinel_for<_It> _Se>
-        _NODISCARD constexpr iter_difference_t<_It> operator()(_It _First, _Se _Last) const
-            noexcept(_Nothrow_distance<_It, _Se>) /* strengthened */ {
-            if constexpr (sized_sentinel_for<_Se, _It>) {
-                return _Last - _First;
-            } else {
-                _Adl_verify_range(_First, _Last);
-                return _Distance_unchecked(_Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)));
-            }
+        // clang-format off
+        template <class _It, sentinel_for<_It> _Se>

```

Comment:
Why, did you drop `input_or_output_iterator` from `_It`.

Ditto below

Reply:
I decided the redundancy (`sentinel_for<S, I>` refines `input_or_output_iterator<I>`) was just noise.

Reply:
I tend to think that we should follow the standard to the letter if there is no cost attached.

Otherwise we end up on a slippery slope where we argue what is obvious. Is this more Casey memoizes the whole standard obvious or random dude on Hackernews obvious

Reply:
Restored.